File: core/src/main/java/org/granchi/hollywood/ReinstateModelExceptionHandler.java
package org.granchi.hollywood;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * ModelExceptionHandler that keeps the same model that threw the exception, discarding the action.
 *
 * @author serandel
 */
public class ReinstateModelExceptionHandler implements ModelExceptionHandler {
    private static final Logger log = LoggerFactory.getLogger(HollywoodApplication.class);

    @Override
    public Model onException(Model model, Action action, Exception exception) {
        log.debug("Reinstating model {}", model);

        return model;
    }
}

File: core/src/main/kotlin/org/granchi/hollywood/ReinstateModelExceptionHandler.kt
package org.granchi.hollywood

import org.slf4j.Logger
import org.slf4j.LoggerFactory

/**
 * [ModelExceptionHandler] that keeps the same [model] that threw the [exception], discarding the [action].
 */
class ReinstateModelExceptionHandler(
        private val log: Logger
        = LoggerFactory.getLogger(ReinstateModelExceptionHandler::class.qualifiedName))
    : ModelExceptionHandler {

    override fun onException(model: Model, action: Action, exception: Exception): Model? {
        log.debug("Reinstating model {}", model)

        return model
    }
}
